### DISCUSSION TITLE
Multiple one-to-many relations on the same model?

### DISCUSSION THREAD

**Discussion by jacobchrismarsh** [prisma/prisma#20887] [Q&A]
### Question

I'm trying to map an employee hierarchy in Prisma, and I'm wondering the best way to do this. I want three things:
1. To see an employee's manager
2. To see their direct reports
3. To see their _indirect_ reports (all employees underneath them in the hierarchy that _aren't_ direct reports)

1 and 2 are pretty easy:
```prisma
model Employee {
  id        Int  @id @default(autoincrement())
  managerId Int?

  manager        Employee?   @relation("ManagerToDirectReports", fields: [managerId], references: [id])
  directReports  Employee[]  @relation("ManagerToDirectReports")
}
```

For 3, I'm struggling a bit. I've tried this:
```prisma
model Employee {
    ...
    indirectReportIds Int[]
    indirectReports   Employee[] @relation("IndirectReports", fields: [indirectReportIds], references: [id])

    // opposite relation
    indirectManagers Employee[] @relation("IndirectReports")
}
```
But that fails with this error: 
```
Error validating: Embedded many-to-many relations are not supported on
Postgres. Please use the syntax defined in https://pris.ly/d/relational-database-many-to-many
```

[The implicit relation docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) _look_ right, but I can't seem to figure it out when it's one table having an implicit relationship with itself. Any idea how I can accomplish this?

**Comment by ludralph** [comment]
Hi @jacobchrismarsh 👋 

Thank you for raising this question.

Mapping an employee hierarchy in Prisma can be achieved using [self-relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations). You've already correctly implemented the one-to-many self-relation for direct reports and managers. However, for indirect reports, you're trying to use an embedded many-to-many relation which is not supported in Postgres as the error message indicates.
Instead, you can use an implicit many-to-many self-relation. Here's an example of how you can define this:
```prisma
model Employee {
  id        Int    @id @default(autoincrement())
  name      String
  managerId Int?

  manager       Employee?  @relation("ManagerToDirectReports", fields: [managerId], references: [id])
  directReports Employee[] @relation("ManagerToDirectReports")

  indirectManagers Employee[] @relation("IndirectReports")
  indirectReports  Employee[] @relation("IndirectReports")
}

```
This relation expresses the following:
- "an employee can be indirectly managed by zero or more employees"
- "an employee can indirectly manage zero or more employees"

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
